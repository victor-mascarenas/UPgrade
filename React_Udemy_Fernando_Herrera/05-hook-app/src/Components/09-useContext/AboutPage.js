import React from 'react';

export const AboutPage = () => {
    return <div>
        <h1>About page</h1>
        <hr/>
    </div>;
};
