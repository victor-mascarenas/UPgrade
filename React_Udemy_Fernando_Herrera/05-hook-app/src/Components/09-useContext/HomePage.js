import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomePage = () => {
    const { user } = useContext(UserContext);

    return <div>
        <h1>Home page</h1>
        <hr/>
        <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>;
};
