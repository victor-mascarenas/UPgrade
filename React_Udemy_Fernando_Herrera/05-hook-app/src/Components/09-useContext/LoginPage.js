import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginPage = () => {
    const {setUser} = useContext(UserContext);

    const buttonOnClick = () => {
        setUser({
            id: 1234, 
            name: 'Victor',
            email: 'vijagama@outlook.es'
        });
    };

    return <div>
        <h1>Login page</h1>
        <hr/>
        <button className='btn btn-primary' onClick={buttonOnClick}>Login</button>
    </div>;
};
