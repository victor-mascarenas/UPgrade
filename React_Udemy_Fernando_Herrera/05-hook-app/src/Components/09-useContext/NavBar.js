import React from 'react';
import {Link, NavLink} from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
            <div className="container-fluid">
                <Link to='#' className='navbar-brand'>useContext</Link>
                <div className="navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <NavLink end='active' to='/' className='nav-link'>Home</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink end='active' to='about' className="nav-link">About</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink end='active' to='login' className="nav-link">Login</NavLink>
                        </li>
                    </ul>
                </div>
            </div>
        </nav>
    );
};
