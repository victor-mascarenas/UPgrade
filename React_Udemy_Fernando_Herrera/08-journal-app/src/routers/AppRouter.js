import React from 'react'
import { BrowserRouter, Route, Routes, Navigate } from 'react-router-dom'
import { JournalScreen } from '../components/journal/JournalScreen'
import { AuthRouter } from './AuthRouter'

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/auth/*' element={<AuthRouter/>}/>
                <Route exact path='/' element={<JournalScreen/>}/>
                <Route path='*' element={<Navigate to='/auth/login'/>}/>
            </Routes>
        </BrowserRouter>
    );
}
