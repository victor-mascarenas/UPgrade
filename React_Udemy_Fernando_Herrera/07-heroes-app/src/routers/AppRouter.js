import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { LoginPage } from '../components/login/LoginPage';

import { DashboardRouter } from './DashboardRouter';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={<LoginPage/>}/>
                <Route path='/*' element={<DashboardRouter/>}/>
            </Routes>
        </BrowserRouter>
    );
};
