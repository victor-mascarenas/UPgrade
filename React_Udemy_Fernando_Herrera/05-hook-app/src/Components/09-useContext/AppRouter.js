import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <div>
                <NavBar/>
                <div className='container'>
                    <Routes>
                        <Route exact path="/" element={<HomePage/>}/>
                        <Route exact path="about" element={<AboutPage/>}/>
                        <Route exact path="login" element={<LoginPage/>}/>
                    </Routes>
                </div>
            </div>
        </BrowserRouter>
    );
};
