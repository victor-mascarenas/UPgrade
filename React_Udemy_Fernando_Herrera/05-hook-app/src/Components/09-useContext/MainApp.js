import React from 'react';
import { AppRouter } from './AppRouter';

export const MainApp = () => {
    /* return <div>
        <h1>Main App</h1>
        <hr/>
        <AppRouter/>
    </div>; */
    return <AppRouter/>;
};
