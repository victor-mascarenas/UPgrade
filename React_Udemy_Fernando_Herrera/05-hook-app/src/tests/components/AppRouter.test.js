import React from 'react';
import {mount} from 'enzyme';
import { AppRouter } from '../../Components/09-useContext/AppRouter';
import { UserContext } from '../../Components/09-useContext/UserContext';

describe('Pruebas en AppRouter', () => {
    const user = {
        id: 1234,
        name: 'Victor',
        email: 'vijagama@outlook.es'
    };
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter/>
        </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});
