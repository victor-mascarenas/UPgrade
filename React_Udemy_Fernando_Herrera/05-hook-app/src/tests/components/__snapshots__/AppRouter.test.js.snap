// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home page
      </h1>
      <hr />
      <pre>
        {
   "id": 1234,
   "name": "Victor",
   "email": "vijagama@outlook.es"
}
      </pre>
    </div>
  </div>
</div>
`;
